lib/validate.js

```diff
@@ -1,82 +1,75 @@
+      'Did you mistype it in package.json?'
+    ]
+  }
+
+  var lines = message.split('\n')
+
+  if (semver.valid(lines[0])) {
+    return []
+  }
+
+  if (lines[0].match(RegExp(options.wip))) {
+    return []
+  }
+
+  if (lines[0].match(RegExp(options.merge))) {
+    return []
+  }
+
+  if (lines[0].match(RegExp(options.fixup))) {
+    return []
+  }
+
+  if (lines[0].match(RegExp(options.squash))) {
+    return []
+  }
 
   var errors = []
 
-  // Can't cancel forEach, so abuse every to stop at first comment
-  message.split(/\n/).every(function (line, index) {
-    if (index === 0) {
-      // Allow tag commits
-      if (semver.valid(line)) {
-        return
-      }
-
-      // Allow merge commits
-      if (/^Merge branch/.test(line) || /^Merge [0-9a-f]+ into [0-9a-f]+/.test(line)) {
-        return
-      }
-      if (line.length === 0) {
-        errors.push('First line (subject) must not be empty')
-      }
-      if (line.length > options.limits.subject) {
-        errors.push('First line (subject) must be no longer than ' +
-          options.limits.subject + ' characters')
-      }
-
-      if (!subjectExceptions.test(line)) {
-        if (options.component && line.indexOf(':') < 1) {
-          errors.push('First line (subject) must indicate the component')
-        }
-        var matchedComponent = line.replace(/:.+/, '')
-        if (options.components.length && matchedComponent &&
-          options.components.indexOf(matchedComponent) === -1) {
-          errors.push("Component invalid, was '" + matchedComponent +
-            "', must be one of these: " + options.components.join(', '))
-        }
-        if (options.component && line.substring(line.indexOf(':') + 1).length < 1) {
-          errors.push('First line (subject) must have a message after the component')
-        }
-      }
-    }
-    if (index === 1 && line.length > 0) {
-      errors.push('Second line must always be empty')
-    }
-    if (index > 1 && line.length > options.limits.other) {
-      errors.push('Commit message line ' + (index + 1) + ' too long: ' +
-        line.length + ' characters, only ' + options.limits.other +
-        ' allowed. Was: ' + line.substring(0, 20) + '[...]')
-    }
-
-    var markerPattern = new RegExp(options.markerPattern, 'i')
-    var actionPattern = new RegExp(options.actionPattern)
-    var ticketPattern = new RegExp(options.ticketPattern)
-
-    // Ticket references
-    if (index > 0 && markerPattern.test(line)) {
-      if (!actionPattern.test(line) && !ticketPattern.test(line)) {
-        errors.push('Invalid ticket reference, must be ' +
-          ticketPattern + ', was: ' + line)
-      }
-    }
-    return true
-  })
+  limits(lines, options, errors)
+
+  if (!options.noTicketReferences) {
+    tickets(lines, options, errors)
+  }
+
+  if (options.style === 'jquery') {
+    jquery(lines, options, errors)
+  }
+
+  if (options.style === 'angular') {
+    angular(lines, options, errors)
```

Comment:
`options.style` must be either 'jquery' or 'angular' at this point, right? Can we then do something like `styles[options.style](...)`?